### DISCUSSION TITLE
Authentication credentials are missing or invalid

### DISCUSSION THREAD

**Discussion by shawnmcpeek** [codemagic-ci-cd/codemagic-docs#2768] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I am trying to make sure my codemagic codesigning is properly configured. I even did this watching a webinar that codemagic put out in 2022 on how to do this process. I uploaded the api key as told. When I try to add ios certificates I am getting this new error:
Failed to fetch certificates:
Authentication credentials are missing or invalid. - Provide a properly configured and signed bearer token, and make sure that it has not expired. Learn more about Generating Tokens for API Requests https://developer.apple.com/go/?id=api-generating-tokens

I get the same error in trying to generate new certificates as well as trying to fetch them. 
I did have a set working before in another attempt months ago, but I don't have that set up any more. I was never asked to make a token json or anything, and that page is even more confusing as to what is wanted than just about anything.

### Expected results

Certificate generated/fetched!

### Actual results

Authentication credentials are missing or invalid. - Provide a properly configured and signed bearer token, and make sure that it has not expired. Learn more about Generating Tokens for API Requests https://developer.apple.com/go/?id=api-generating-tokens

**Comment by dam-ease** [comment]
Hello @shawnmcpeek.
At what point do you get this error? Is it when fetching certificates from Apple developer portal in code signing identities section?
This might be an issue with your .p8 file, possibly due to incorrect permissions or corrupt content. My suggestion would be to generate a new API key in your Apple Developer account from App Store Connect > Users and Access > Keys, making sure to grant yourself **App Manager or Admin permissions**. Take note of the KeyID as well as the Issuer ID. Once that's done, simply upload the new key with the raw content of the key, no adjustments (without any spaces or base64 encoding). I'm sure this solution will do the trick, but please do let me know if you need any further assistance.

**Comment by shawnmcpeek** [comment]
Thank you for the assistance. I tried what you said and it still gives me
the same issue.

On Thu, Oct 10, 2024 at 5:51 AM Damilola Alimi ***@***.***>
wrote:

> Hello @shawnmcpeek <https://github.com/shawnmcpeek>.
> At what point do you get this error? Is it when fetching certificates from
> Apple developer portal in code signing identities section?
> This might be an issue with your .p8 file, possibly due to incorrect
> permissions or corrupt content. My suggestion would be to generate a new
> API key in your Apple Developer account from App Store Connect > Users and
> Access > Keys, making sure to grant yourself *App Manager or Admin
> permissions*. Take note of the KeyID as well as the Issuer ID. Once
> that's done, simply upload the new key with the raw content of the key, no
> adjustments (without any spaces or base64 encoding). I'm sure this solution
> will do the trick, but please do let me know if you need any further
> assistance.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/codemagic-ci-cd/discussions/2768#discussioncomment-10903385>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6AS2EG5TQMCYQPAIA7HUXDZ2ZS4DAVCNFSM6AAAAABPOZN4POVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOJQGMZTQNI>
> .
> You are receiving this because you were mentioned.Message ID:
> <codemagic-ci-cd/codemagic-docs/repo-discussions/2768/comments/10903385@
> github.com>
>


-- 
Vivat Jesus!
Shawn McPeek
PGK FS Council 15857
PFN Assembly 94

**Reply by dam-ease** [reply]
Oh, that's strange!
Can you check [here](https://appstoreconnect.apple.com/access/integrations/api) to confirm if your App Store Connect API key has been revoked? If it isn't try recreating a new one and add it to Codemagic (with Admin or App manager role)
Check also in the following [link](https://appstoreconnect.apple.com/agreements/#/) if there's any pending agreements that need to be signed

**Comment by shawnmcpeek** [comment]
Both API keys I have didn't show revoked or anything.

I didn't see any agreements that are needing to be signed. Everything looks
correct and up to date

**Reply by shawnmcpeek** [reply]
I have also created new API keys and same results. I would be happy to give someone team access to apple and let them try and hopefully fix whatever I did wrong.

**Reply by dam-ease** [reply]
Hello @shawnmcpeek.
Please let's schedule a screen sharing call at your convenience so we can look into this.
https://calendly.com/damilola-nevercode/30min
